diff -ruN a/api/BUILD.bazel b/api/BUILD.bazel
--- a/api/BUILD.bazel	2026-01-16 15:19:23
+++ b/api/BUILD.bazel	2026-01-16 15:19:32
@@ -9,9 +9,9 @@
     ]),
     visibility = ["//visibility:public"],
     deps = [
-        artifact("com.google.code.findbugs:jsr305"),
-        artifact("com.google.errorprone:error_prone_annotations"),
-        artifact("com.google.guava:failureaccess"),  # future transitive dep of Guava. See #5214
-        artifact("com.google.guava:guava"),
+        artifact("com.google.code.findbugs:jsr305", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.errorprone:error_prone_annotations", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.guava:failureaccess", repository_name = "contrib_rules_jvm_deps"),  # future transitive dep of Guava. See #5214
+        artifact("com.google.guava:guava", repository_name = "contrib_rules_jvm_deps"),
     ],
 )
diff -ruN a/compiler/BUILD.bazel b/compiler/BUILD.bazel
--- a/compiler/BUILD.bazel	2026-01-16 15:19:24
+++ b/compiler/BUILD.bazel	2026-01-16 15:19:57
@@ -24,9 +24,9 @@
         "//api",
         "//protobuf",
         "//stub",
-        artifact("com.google.code.findbugs:jsr305"),
-        artifact("com.google.guava:guava"),
-        "@com_google_protobuf//:protobuf_java",
+        artifact("com.google.code.findbugs:jsr305", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.guava:guava", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.protobuf:protobuf-java", repository_name = "contrib_rules_jvm_deps"),
     ],
 )
 
@@ -36,8 +36,8 @@
         "//api",
         "//protobuf-lite",
         "//stub",
-        artifact("com.google.code.findbugs:jsr305"),
-        artifact("com.google.guava:guava"),
+        artifact("com.google.code.findbugs:jsr305", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.guava:guava", repository_name = "contrib_rules_jvm_deps"),
     ],
 )
 
diff -ruN a/netty/BUILD.bazel b/netty/BUILD.bazel
--- a/netty/BUILD.bazel	2026-01-16 15:19:23
+++ b/netty/BUILD.bazel	2026-01-16 15:19:32
@@ -13,22 +13,22 @@
     deps = [
         "//api",
         "//core:internal",
-        artifact("com.google.code.findbugs:jsr305"),
-        artifact("com.google.errorprone:error_prone_annotations"),
-        artifact("com.google.guava:guava"),
-        artifact("io.netty:netty-buffer"),
-        artifact("io.netty:netty-codec"),
-        artifact("io.netty:netty-codec-http"),
-        artifact("io.netty:netty-codec-http2"),
-        artifact("io.netty:netty-codec-socks"),
-        artifact("io.netty:netty-common"),
-        artifact("io.netty:netty-handler"),
-        artifact("io.netty:netty-handler-proxy"),
-        artifact("io.netty:netty-resolver"),
-        artifact("io.netty:netty-transport"),
-        artifact("io.netty:netty-transport-native-unix-common"),
-        artifact("io.perfmark:perfmark-api"),
-        artifact("org.codehaus.mojo:animal-sniffer-annotations"),
+        artifact("com.google.code.findbugs:jsr305", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.errorprone:error_prone_annotations", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.guava:guava", repository_name = "contrib_rules_jvm_deps"),
+        artifact("io.netty:netty-buffer", repository_name = "contrib_rules_jvm_deps"),
+        artifact("io.netty:netty-codec", repository_name = "contrib_rules_jvm_deps"),
+        artifact("io.netty:netty-codec-http", repository_name = "contrib_rules_jvm_deps"),
+        artifact("io.netty:netty-codec-http2", repository_name = "contrib_rules_jvm_deps"),
+        artifact("io.netty:netty-codec-socks", repository_name = "contrib_rules_jvm_deps"),
+        artifact("io.netty:netty-common", repository_name = "contrib_rules_jvm_deps"),
+        artifact("io.netty:netty-handler", repository_name = "contrib_rules_jvm_deps"),
+        artifact("io.netty:netty-handler-proxy", repository_name = "contrib_rules_jvm_deps"),
+        artifact("io.netty:netty-resolver", repository_name = "contrib_rules_jvm_deps"),
+        artifact("io.netty:netty-transport", repository_name = "contrib_rules_jvm_deps"),
+        artifact("io.netty:netty-transport-native-unix-common", repository_name = "contrib_rules_jvm_deps"),
+        artifact("io.perfmark:perfmark-api", repository_name = "contrib_rules_jvm_deps"),
+        artifact("org.codehaus.mojo:animal-sniffer-annotations", repository_name = "contrib_rules_jvm_deps"),
     ],
 )
 
diff -ruN a/protobuf/BUILD.bazel b/protobuf/BUILD.bazel
--- a/protobuf/BUILD.bazel	2026-01-16 15:19:24
+++ b/protobuf/BUILD.bazel	2026-01-16 15:20:11
@@ -10,10 +10,10 @@
     deps = [
         "//api",
         "//protobuf-lite",
-        "@com_google_protobuf//:protobuf_java",
-        artifact("com.google.api.grpc:proto-google-common-protos"),
-        artifact("com.google.code.findbugs:jsr305"),
-        artifact("com.google.errorprone:error_prone_annotations"),
-        artifact("com.google.guava:guava"),
+        artifact("com.google.protobuf:protobuf-java", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.api.grpc:proto-google-common-protos", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.code.findbugs:jsr305", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.errorprone:error_prone_annotations", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.guava:guava", repository_name = "contrib_rules_jvm_deps"),
     ],
 )
diff -ruN a/protobuf-lite/BUILD.bazel b/protobuf-lite/BUILD.bazel
--- a/protobuf-lite/BUILD.bazel	2026-01-16 15:19:24
+++ b/protobuf-lite/BUILD.bazel	2026-01-16 15:19:43
@@ -9,12 +9,10 @@
     visibility = ["//visibility:public"],
     deps = [
         "//api",
-        artifact("com.google.code.findbugs:jsr305"),
-        artifact("com.google.guava:guava"),
-    ] + select({
-        ":android": ["@com_google_protobuf//:protobuf_javalite"],
-        "//conditions:default": ["@com_google_protobuf//:protobuf_java"],
-    }),
+        artifact("com.google.code.findbugs:jsr305", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.guava:guava", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.protobuf:protobuf-java", repository_name = "contrib_rules_jvm_deps"),
+    ],
 )
 
 # This config is not fully-reliable. If it breaks, it is probably because you
diff -ruN a/stub/BUILD.bazel b/stub/BUILD.bazel
--- a/stub/BUILD.bazel	2026-01-16 15:19:23
+++ b/stub/BUILD.bazel	2026-01-16 15:19:32
@@ -10,9 +10,17 @@
     deps = [
         "//api",
         "//context",
-        artifact("com.google.code.findbugs:jsr305"),
-        artifact("com.google.errorprone:error_prone_annotations"),
-        artifact("com.google.guava:guava"),
-        artifact("org.codehaus.mojo:animal-sniffer-annotations"),
+        artifact("com.google.code.findbugs:jsr305", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.errorprone:error_prone_annotations", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.guava:guava", repository_name = "contrib_rules_jvm_deps"),
+        artifact("org.codehaus.mojo:animal-sniffer-annotations", repository_name = "contrib_rules_jvm_deps"),
     ],
 )
+
+# Backwards compatibility shim for @Generated annotation
+java_library(
+    name = "javax_annotation",
+    neverlink = 1,  # @Generated is source-retention
+    visibility = ["//visibility:public"],
+    exports = [artifact("org.apache.tomcat:annotations-api", repository_name = "contrib_rules_jvm_deps")],
+)
